File: Medium/2026-01-21-3315-Construct-the-Minimum-Bitwise-Array-II/solution.java
class Solution {
    public int[] constructArray(int[] nums) {
        int n = nums.length;
        int[] ans = new int[n];
        
        for (int i = 0; i < n; i++) {
            int num = nums[i];
            ans[i] = -1; // Default value
            
            // Check if we can find a valid ans[i]
            for (int j = 0; j <= num; j++) {
                if ((j | (j + 1)) == num) {
                    ans[i] = j;
                    break;
                }
            }
        }
        
        return ans;
    }
}

File: Medium/2026-01-21-3315-Construct-the-Minimum-Bitwise-Array-II/solution.js
function constructArray(nums) {
    const ans = [];
    
    for (const num of nums) {
        let found = false;
        for (let i = 0; i <= num; i++) {
            if ((i | (i + 1)) === num) {
                ans.push(i);
                found = true;
                break;
            }
        }
        if (!found) {
            ans.push(-1);
        }
    }
    
    return ans;
}